### ISSUE TITLE
Database disk image is malformed,

### ISSUE THREAD

**Issue by tuceturan** [simolus3/sqlite3.dart#193]
Hello,

I'm using sqlite in my android app.When doing multi insert  ı use ;
I use existing database in sqlite3 

 final _db = .sqlite3.open(fileDb.path);
 var stmt = _db.prepare(...)
 stmt.dispose();
 like that but sometimes its throws error "while executing statement, database disk image is malformed"

**Comment by simolus3** [comment]
Are you continuing to use the existing database natively? If so, does that also report errors about malformed database images?

I think this might be a concurrency issue, but we're compiling sqlite with thread-safe options so it should be fine. I wonder if the native side might be using the database without thread safety, but it seems unlikely when using the default Android APIs.

**Comment by tuceturan** [comment]
what do you mean for " I wonder if the native side might be using the database without thread safety" sorry ı couldn't understand

**Comment by simolus3** [comment]
I don't think this is the source of the problem, but if two threads are using the same sqlite3 database _and_ that database hasn't been configured to be used with thread safety (an option in sqlite3), that might corrupt the database. But again, this is unlikely the problem here.

Do you have a way to check whether you're also getting the malformed database errors natively, or are you only accessing the database in Dart now?

**Comment by tuceturan** [comment]
just accessting database in dart

**Comment by Den-creator** [comment]
have the similar issue

**Comment by simolus3** [comment]
@Den-creator Could you also describe your setup in a bit more detail (e.g. does this only happen on some platforms/operating systems, are you opening multiple connections to the same database file (even in different threads), did you notice anything else that seems to cause this problem or make it more likely)? At the moment I don't really have enough information to investigate this, so everything helps.

**Comment by Den-creator** [comment]
@simolus3 looks like the issue has gone when upgraded to the latest available version, I will keep you informed if issue will raise again.

**Comment by bendangelo** [comment]
Hi, I've been able to reproduce this by using a sqlite db created by the Sequel gem: https://github.com/jeremyevans/sequel

Use the db file created like this:
```
DB = Sequel.connect('sqlite://blog.db')

DB.create_table :items do
  primary_key :id
  String :name
  Float :price
end

items = DB[:items] # Create a dataset

# Populate the table
items.insert(name: 'abc', price: rand * 100)
items.insert(name: 'def', price: rand * 100)
items.insert(name: 'ghi', price: rand * 100)

# Print out the number of records
puts "Item count: #{items.count}"

# Print out the average price
puts "The average price is: #{items.avg(:price)}"
```

You'll get the malformed error.

**Comment by simolus3** [comment]
I couldn't reproduce the error. I have this `Gemfile.lock`:

```
GEM
  remote: https://rubygems.org/
  specs:
    bigdecimal (3.1.8)
    mini_portile2 (2.8.7)
    sequel (5.84.0)
      bigdecimal
    sqlite3 (2.0.4)
      mini_portile2 (~> 2.8.0)
    sqlite3 (2.0.4-x86_64-linux-gnu)

PLATFORMS
  ruby
  x86_64-linux

DEPENDENCIES
  sequel
  sqlite3

BUNDLED WITH
   2.5.11
```

With that, your snippet (well, with a `require 'sequel'` at the top) generated a `blog.db` that I was able to open like this:

```dart
import 'package:sqlite3/sqlite3.dart';

void main() {
  final dbtest = sqlite3.open('/tmp/test/blog.db');
  print(dbtest.select('select * from items'));
}
```

**Comment by bendangelo** [comment]
I found the issue. It had to do with the left over -mal -shm files. Those have to be deleted.
